Weather/WeatherService.swift

```diff
@@ -37,58 +142,500 @@ struct WeatherData: Codable {
+                description = ((FutureForcast?.day?.phrase_32char)!)
+                let units: String = Preferences[.units]
+                if Preferences[.show_description]{
+                    switch units {
+                    case "celsius":
+                            if description.count > 19{
+                                name = ((FutureForcast?.day?.daypart_name)! + " High: " + String(Int(Double(FutureForcast?.metric?.maxTemp ?? 0))) + "°" + ("  Low: " + String(Int(Double(FutureForcast?.metric?.minTemp ?? 0)))) + "°" )
+                            }else{
+                                name = ((FutureForcast?.day?.daypart_name)! + "  " + String(Int(Double(FutureForcast?.metric?.maxTemp ?? 0))) + "°")
+                            }
+                    default:
+                        name = ((FutureForcast?.day?.daypart_name)! + "  " + String(Int(Double(FutureForcast?.imperial?.maxTemp ?? 0))) + "°")
+                    }
+                    if !(description.count > 19){
+                        temp = ("Low: " + String(Int(Double(FutureForcast?.metric?.minTemp ?? 0))))
+                        temp_F = ("Low: " + String(Int(Double(FutureForcast?.imperial?.minTemp ?? 0))))
+                    }else{
+                        temp = ""
+                        temp_F = ""
+                    }
+                }else{
+                    name = ((FutureForcast?.day?.daypart_name)!)
+                    temp = ("High: " + String(Int(Double(FutureForcast?.metric?.maxTemp ?? 0))) + "°" + ("Low: " + String(Int(Double(FutureForcast?.metric?.minTemp ?? 0)))) + "°" )
+                    temp_F = ("High: " + String(Int(Double(FutureForcast?.imperial?.maxTemp ?? 0))) + "°" + ("Low: " + String(Int(Double(FutureForcast?.imperial?.minTemp ?? 0)))) + "°" )
+                }
+                DisplayIcon = FutureForcast?.day?.icon_cd!
+
+            }else{
+                temp = String(Int(Double(observation?.metric?.temp ?? 0)))
+                temp_F = String(Int(Double(observation?.imperial?.temp ?? 0)))
+                description = (observation?.wx_phrase)!
+                DisplayIcon = observation?.wx_icon!
             }
+            print(String(DisplayIcon!))
+            if Preferences[.IconStyle] == "Outlined"{
+                Icon = " -" + String(DisplayIcon!)
+            }else if Preferences[.IconStyle] == "Filled"{
+                Icon = "2-" + String(DisplayIcon!)
+            }else if Preferences[.IconStyle] == "Default"{
+                if DisplayIcon == 011 || DisplayIcon == 012 || DisplayIcon == 013 || DisplayIcon == 014 || DisplayIcon == 016 || DisplayIcon == 019 || DisplayIcon == 020 || DisplayIcon == 021 || DisplayIcon == 022 || DisplayIcon == 026 || DisplayIcon == 027 || DisplayIcon == 028 || DisplayIcon == 029 || DisplayIcon == 030 || DisplayIcon == 031 || DisplayIcon == 032 || DisplayIcon == 036 || DisplayIcon == 037 || DisplayIcon == 038 || DisplayIcon == 039 || DisplayIcon == 040 || DisplayIcon == 047{
```

Comment:
Checks if icon is available in default icon pack, if not it displays outlined icon pack.